rust/tool/src/utils.rs

```diff
@@ -0,0 +1,55 @@
+use std::fs;
+use std::io::Write;
+use std::os::unix::fs::OpenOptionsExt;
+use std::path::{Path, PathBuf};
+
+use anyhow::{Context, Result};
+use tempfile::TempDir;
+
+/// A temporary directory that enables creating secure temporary files in it.
+///
+/// The temporary directory has three useful properties:
+///
+/// - Files are created with mode 0o600, so that they are only accessible by the current user.
+/// - Files are named and not entirely ephemeral (unlike a real temporary file).
+/// - The directory and its children are cleaned up (i.e. deleted) when the variable that holds the
+///   directory goes out of scope.
+///
+/// This lightly protects against an attacker without root access from modifying files undetected.
```

Comment:
So does it protect or not? If not, what does it not protect against?

Reply:
I clarified this a tiny bit. For more detail, I'd have to do some more extensive threat modeling.